packages/browser/test/e2e/test/builtins.test.js

```diff
@@ -0,0 +1,322 @@
+      await getSentryRequest(page, () => {
+        function foo() {
+          window.capturedCtx = this;
+          // Capturing a message to trigger `getSentryRequest`.
+          Sentry.captureMessage('foo');
+        }
+        requestAnimationFrame(foo.bind({ magicNumber: 42 }));
+      });
+
+      const capturedCtx = await page.evaluate(() => window.capturedCtx);
+
+      expect(capturedCtx.magicNumber).toBe(42);
+    });
+  });
+
+  test('should capture exceptions from XMLHttpRequest event handlers (e.g. onreadystatechange)', async ({ page }) => {
+    // Note: Using `getSentryRequest` here as it waits for the callback inside `requestAnimationFrame` by default.
+    const requestData = await getSentryRequest(page, () => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', '/base/subjects/example.json');
+      // intentionally assign event handlers *after* open, since this is what jQuery does
+      xhr.onreadystatechange = function wat() {
+        // replace onreadystatechange with no-op so exception doesn't
+        // fire more than once as XHR changes loading state
+        xhr.onreadystatechange = function() {};
+        // eslint-disable-next-line no-undef
+        foo();
+        // Capturing a message to trigger `getSentryRequest`.
+        Sentry.captureMessage('foo');
+      };
+      xhr.send();
+    });
+
+    expect(requestData.exception.values[0].value).toMatch(/foo/);
+    expect(requestData.exception.values[0].mechanism).toMatchObject({
+      type: 'instrument',
+      handled: true,
+      data: {
+        function: 'onreadystatechange',
+      },
+    });
+  });
+
+  test('should not call XMLHttpRequest onreadystatechange more than once per state', async ({ page }) => {
+    // Note: Using `getSentryRequest` here as it waits for the callback inside `requestAnimationFrame` by default.
+    await getSentryRequest(page, () => {
+      window.calls = {};
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', '/base/subjects/example.json');
+      xhr.onreadystatechange = function wat() {
+        window.calls[xhr.readyState] = window.calls[xhr.readyState] ? window.calls[xhr.readyState] + 1 : 1;
+        if (xhr.readyState === 4) {
+          // Capturing a message to trigger `getSentryRequest`.
+          Sentry.captureMessage('foo');
+        }
+      };
+      xhr.send();
+    });
+
+    const calls = await page.evaluate(() => window.calls);
+
+    // eslint-disable-next-line guard-for-in
+    for (const state in calls) {
+      expect(calls[state]).toBe(1);
+    }
+
+    expect(Object.keys(calls).length).toBeGreaterThanOrEqual(3);
+    expect(Object.keys(calls).length).toBeLessThanOrEqual(4);
+  });
+
+  test(`should capture built-in's mechanism type as instrument`, async ({ page }) => {
+    const requestData = await getSentryRequest(page, () => {
+      setTimeout(function() {
+        // eslint-disable-next-line no-undef
```

Comment:
Can we think of a better way to apply this?
Having `// eslint-disable-next-line no-undef` repeat all over the place doesn't seem nice. I don't know weather extracting this in a function like this would work (or something like that):
```javascript
const foo = () => {
    // eslint-disable-next-line no-undef
    return undefined
}
```

Reply:
I'll try to move most of the `injection` logic into fixtures / subjects. Can define a specific eslint rule set for subjects folder then.